src/sagemaker/cli/compatibility/v2/modifiers/serde.py

```diff
@@ -0,0 +1,413 @@
+
+    def node_should_be_modified(self, node):
+        """Checks if the ``ast.Name`` node identifies a SerDe object.
+
+        This looks for the following objects:
+
+        - ``sagemaker.predictor.csv_serializer``
+        - ``sagemaker.predictor.json_serializer``
+        - ``sagemaker.predictor.npy_serializer``
+        - ``sagemaker.predictor.csv_deserializer``
+        - ``sagemaker.predictor.json_deserializer``
+        - ``sagemaker.predictor.numpy_deserializer``
+
+        Args:
+            node (ast.Call): a node that represents a function call. For more,
+                see https://docs.python.org/3/library/ast.html#abstract-grammar.
+
+        Returns:
+            bool: If the ``ast.Call`` instantiates a SerDe class.
+        """
+        name = node.id if isinstance(node, ast.Name) else node.attr
+        return name in OLD_OBJECT_NAMES
+
+    def modify_node(self, node):
+        """Replaces the ``ast.Name`` node with a ``ast.Call`` node that
+        instantiates a class available in version 2.0 and later of the Python SDK:
+
+        - ``sagemaker.serializers.CSVSerializer()``
+        - ``sagemaker.serializers.JSONSerializer()``
+        - ``sagemaker.serializers.NumpySerializer()``
+        - ``sagemaker.deserializers.CSVDeserializer()``
+        - ``sagemaker.deserializers.JSONDeserializer()``
+        - ``sagemaker.deserializers.NumpyDeserializer()``
+
+        The ``sagemaker`` prefix is excluded from the returned node.
+
+        Args:
+            node (ast.Name): a node that represents a Python identifier.
+
+        Returns:
+            ast.Call: a node that represents the instantiation of a SerDe object.
+        """
+        object_name = node.id if isinstance(node, ast.Name) else node.attr
+        new_class_name = OLD_OBJECT_NAME_TO_NEW_CLASS_NAME[object_name]
+        namespace_name = NEW_CLASS_NAME_TO_NAMESPACES[new_class_name][0]
+        subpackage_name = namespace_name[namespace_name.find(".") + 1 :]
+        assert subpackage_name in {"serializers", "deserializers"}
+        return pasta.parse("%s.%s()" % (subpackage_name, new_class_name)).body[0].value
+
+
+class SerdeImportFromPredictorRenamer(Modifier):
+    """A class to update import statements starting with ``from sagemaker.predictor``."""
+
+    def node_should_be_modified(self, node):
+        """Checks if the import statement imports a SerDe from the
+        ``sagemaker.predictor`` module.
+
+        Args:
+            node (ast.ImportFrom): a node that represents a ``from ... import ... `` statement.
+                For more, see https://docs.python.org/3/library/ast.html#abstract-grammar.
+
+        Returns:
+            bool: True if and only if the ``ast.ImportFrom`` imports a SerDe
+                from the ``sagemaker.predictor`` module.
+        """
+        return node.module == "sagemaker.predictor" and any(
+            [name.name in (OLD_CLASS_NAMES | OLD_OBJECT_NAMES) for name in node.names]
```

Comment:
no need to construct a `list` (i.e. remove the `[` `]`) as generator expressions are just fine for `any`:

```
[ins] In [1]: any(x for x in [False, True])
Out[1]: True
```

Reply:
Did not know that. That's useful to know. Fixed